tests/attention/test_asymmetric_kv_decode.py

```
@@ -0,0 +1,382 @@
+        dtype=torch.float16,
+    )
+    with pytest.raises(ValueError, match="v_data_type"):
+        wrapper.run(q, (k_cache, v_cache_wrong))
+
+
+def test_asymmetric_tensor_cores_raises():
+    workspace_buffer = torch.empty(32 * 1024 * 1024, dtype=torch.int8, device="cuda:0")
+    wrapper = flashinfer.decode.BatchDecodeWithPagedKVCacheWrapper(
+        workspace_buffer, "NHD", use_tensor_cores=True
+    )
+    with pytest.raises(NotImplementedError, match="[Aa]symmetric"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)